discovery.go

```diff
@@ -0,0 +1,63 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	pubsub "github.com/libp2p/go-floodsub"
+	host "github.com/libp2p/go-libp2p-host"
+	pstore "github.com/libp2p/go-libp2p-peerstore"
+)
+
+type Discovery interface {
+	Advertise(ctx context.Context, shardID ShardIDType) error
```

Comment:
Should the advertised `topic` be only specific to `ShardIDType` here? I was thinking if this `Discovery` can be more general. Like, maybe we can make it support the discovery of beacon chain nodes, or even no topic at all.

Reply:
Totally agree. That was my plan but we can not support general topic discovery with global shard preference table. I will change it.